### ISSUE TITLE
Proposal: allow BaseExperiment to accept experiment configuration directly

### ISSUE THREAD

**Issue by ema-pe** [FFede0/RL4CC#10]
Currently the only way to pass the experiment configuration to BaseExperiment is to use the `exp_config_file` argument in the constructor. The proposal is to add a new argument called `exp_config` of type dict. This argument, if not None, will contain the full experiment configuration as expected with `exp_config_file`. The difference is that with `exp_config` there is no need to read a JSON file from disk.

https://github.com/FFede0/RL4CC/blob/ac2c8e2891e166ca75eca6934d8833d7da039aac/experiments/base_experiment.py#L26-L35

What about the `env_config_file` and `ray_config_file` fields in the `exp_config` dictionary? I propose to add two more fields:

* `env_config`: a dictionary, if not None, must contain a full environment configuration as expected when reading the `env_config_file`;
* `ray_config`: a dictionary, if not None, must contain a complete ray configuration as expected when reading the `ray_config_file`;

If the user gives to RL4CC an experiment configuration with both `env_config_file` and `env_config`, it will throw an error. The same behaves with `ray_config_file` and `ray_config`.

If the user passes both the `exp_config_file` and `exp_config` arguments to BaseExperiment, the constructor will throw an error.

Examples of valid experiment configurations are:

<details>
<summary>exp_config given as dictionary, with env_config and ray_config given as files</summary>

```python
{
    "algorithm": "PPO",
    "env_config_file": "env_config.json",
    "ray_config_file": "ray_config.json",
    "stopping_criteria": {
        "max_iterations": 1
    }
}
```
</details>

<details>
<summary>exp_config given as dictionary, with env_config given as dictionary and ray_config given as files</summary>

```python
{
    "algorithm": "PPO",
    "env_config": {
        "env_name": "TrafficManagementEnv",
        "min_time": 0,
        "max_time": 100,
        "time_step": 1,
        "scenario": "scenario1"
     },
    "ray_config_file": "ray_config.json",
    "stopping_criteria": {
        "max_iterations": 1
    }
}
```
</details>

<details>
<summary>exp_config given as dictionary, with env_config and ray_config given as dictionaries</summary>

```python
{
    "algorithm": "PPO",
    "env_config": {
        "env_name": "TrafficManagementEnv",
        "min_time": 0,
        "max_time": 100,
        "time_step": 1,
        "scenario": "scenario1"
     },
    "ray_config_file": {
        "framework": "torch",
        "callbacks": {
            "callbacks_class": "RL4CC.callbacks.base_callbacks.BaseCallbacks"
        },
        "rollouts": {
            "num_rollout_workers": 1
        },
    },
    "stopping_criteria": {
        "max_iterations": 1
    }
}
```
</details>

<details>
<summary>exp_config given as JSON file, with env_config and ray_config given as dictionaries</summary>

```bash
$ cat exp_config.json
{
    "algorithm": "PPO",
    "env_config": {
        "env_name": "TrafficManagementEnv",
        "min_time": 0,
        "max_time": 100,
        "time_step": 1,
        "scenario": "scenario1"
     },
    "ray_config_file": {
        "framework": "torch",
        "callbacks": {
            "callbacks_class": "RL4CC.callbacks.base_callbacks.BaseCallbacks"
        },
        "rollouts": {
            "num_rollout_workers": 1
        },
    },
    "stopping_criteria": {
        "max_iterations": 1
    }
}
```
</details>

**Reason for the proposal**: I need to run a lot of experiments with small changes between each one. So it would be faster to dynamically generate the experiment configuration and pass it to RL4CC, rather than writing the configuration to a temporary file and passing the path to RL4CC.

**Impact and implementation**: minor changes in BaseExperiment, all backwards compatible.

What do you think? If accepted, I can try to implement the feature.

**Comment by FFede0** [comment]
Accepted. Please, implement it branching from `test` and not from `main`, so that we follow the guidelines on contributions that were provided at the beginning.

**Comment by ema-pe** [comment]
> Accepted. Please, implement it branching from `test` and not from `main`, so that we follow the guidelines on contributions that were provided at the beginning.

I can't find the email with the guidelines, I think I've deleted it. For future contributors and reference, can you put please the guidelines in a `CONTRIBUTING.md` file in the root directory of the project?

Edit: I found it!

**Comment by FFede0** [comment]
Added a “how to contribute” section to the README.md in the main branch

**Comment by FFede0** [comment]
Closed by #13
